src/overlays/actors/ovl_Obj_Takaraya_Wall/z_obj_takaraya_wall.c

```diff
@@ -30,39 +50,450 @@ ActorInit Obj_Takaraya_Wall_InitVars = {
+        }
+    }
+}
+
+void ObjTakarayaWall_Manage(ObjTakarayaWall* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    ColliderCylinder* collider;
+    s32 playerRowBehind;
+    s32 playerRow;
+    s32 playerColumn;
+    s32 columnColliderActive;
+    s32 rowColliderActive;
+    s32 playerSpaceXPos;
+    s32 playerSpaceZPos;
+    s32 i;
+    s32 j;
+    s32 playerColumnRight;
+    s32 playerColumnLeft;
+    s32 playerRowInFront;
+
+    rowColliderActive = false;
+    columnColliderActive = false;
+    playerRowBehind = playerRowInFront = -1;
+    playerColumnLeft = playerColumnRight = -1;
+
+    ObjTakarayaWall_PosToSpace(&player->actor.world.pos, &playerRow, &playerColumn);
+    playerSpaceXPos = (((s32)player->actor.world.pos.x + (15 * 120)) % 120) - 60;
+    playerSpaceZPos = (((s32)player->actor.world.pos.z + (1 * 120)) % 120) - 60;
+
+    if (playerSpaceXPos > 20) {
+        playerRowBehind = playerRow + 1;
+    } else if (playerSpaceXPos < -20) {
+        playerRowInFront = playerRow - 1;
+    }
+
+    if (playerSpaceZPos > 20) {
+        playerColumnLeft = playerColumn + 1;
+    } else if (playerSpaceZPos < -20) {
+        playerColumnRight = playerColumn - 1;
+    }
+
+    for (i = 0; i < TAKARAYA_WALL_ROWS; i++) {
+        for (j = 0; j < TAKARAYA_WALL_COLUMNS; j++) {
+            if (sTakarayaWallWallHeights[i][j] >= 0.0f) {
+                if (((j == playerColumn) && ((i == playerRowInFront) || (i == playerRowBehind))) ||
+                    ((i == playerRow) && ((j == playerColumnRight) || (j == playerColumnLeft)))) {
+                    if (Math_StepToF(&sTakarayaWallWallHeights[i][j], 120.0f, 15.0f)) {
+                        sTakarayaWallSpaceStates[i][j] = TAKARAYA_WALL_INACTIVE;
+                    } else {
+                        sTakarayaWallSpaceStates[i][j] = TAKARAYA_WALL_RISING;
+                    }
+                    if (j == playerColumn) {
+                        columnColliderActive = true;
+                    } else if (i == playerRow) {
+                        rowColliderActive = true;
+                    }
+                } else if ((i != playerRow) || (j != playerColumn)) {
+                    Math_SmoothStepToF(&sTakarayaWallWallHeights[i][j], 0.0f, 0.2f, 5.0f, 1.0f);
+                    sTakarayaWallSpaceStates[i][j] = TAKARAYA_WALL_FALLING;
+                }
+            }
+        }
+    }
+
+    collider = &this->columnCollider;
+    collider->dim.radius = 200;
+    if (!columnColliderActive) {
+        collider->dim.pos.y = (s32)this->actor.world.pos.y - 240;
+    } else {
+        collider->dim.pos.y = (s32)this->actor.world.pos.y;
+        collider->dim.pos.z = (s32)player->actor.world.pos.z;
+        if (playerSpaceXPos > 0) {
+            collider->dim.pos.x = ((((s32)player->actor.world.pos.x + (14 * 120)) / 120) * 120) - 1380;
+        } else {
+            collider->dim.pos.x = ((((s32)player->actor.world.pos.x + (14 * 120)) / 120) * 120) - 1860;
```

Comment:
```suggestion
            collider->dim.pos.x = ((((s32)player->actor.world.pos.x - -1680) / 120) * 120) - 1380;
        } else {
            collider->dim.pos.x = ((((s32)player->actor.world.pos.x - -1680) / 120) * 120) - 1860;
```

more things that should be world space coordinates